### ISSUE TITLE
Opening Backpack and Slimefun Guide at the same time causes unwanted behavior

### ISSUE THREAD

**Issue by JustAHuman-xD** [Slimefun/Slimefun4#4131]
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

(See reproduction steps and expected behavior) This bug can lead to mis use of the backpack, dupe bugs, etc. I have confirmed that this exists at least and I have been told that you can dupe items by allowing another player to upgrade the dropped backpack while you have it open, and then you both take out items.

### 📑 Reproduction Steps

1. Get a slimefun guide
2. Get a backpack
3. Place slimefun guide in off hadn
4. Place backpack in main hand
5. Right Click

Observe that you can drop the backpack, put anything in the backpack, etc.

### 💡 Expected Behavior

I expect either the backpack to not open or if it does open to be properly recognized as a back pack and have all the correct handlers.

### 📜 Server Log

https://paste.gg/p/anonymous/70c3e68a275b46ed922e5be3693a3dc6

### 📂 `/error-reports/` folder

(N/A)

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.20.x

### ⭐ Slimefun version

This Server uses the following setup of Slimefun:
Paper git-Paper-405 (MC: 1.20.4)
Slimefun Dev - 1136
Metrics-Module #29
Java 17
Installed Addons: (33)
  MiniBlocks vDEV - 4 (git f7a34b6e)
  Networks vDEV - 50 (git 3de3c9d6)
  CrystamaeHistoria vDEV - 81 (git 0c0efa95)
  AureliumSkills vBeta 1.3.24
  RelicsOfCthonia vDev - 12
  FoxyMachines vDEV - 67 (git 0028aa93)
  Quaptics vDev - 1
  FNAmplifications vDev - 103
  MetaBuilding v0
  Furnished v1.0
  MetaArrows v0
  MetaRank v0.2
  SlimeHUD vDev - 1
  InfinityExpansion v1
  SFCalc vDev - 38
  Aureliumeta v0
  MetaEvents v0
  SlimeTinker vDEV - 84 (git 2c9f6155)
  SpiritsUnchained vDev - 24
  TranscEndence vDEV - 41 (git 98d7d47e)
  Galactifun vDEV - 76 (git aa6414a8)
  MetaStory v0
  CMI v9.6.10.0
  FluffyMachines vDev - 4
  MetaMachines v0.7
  Cultivation v1.0.0
  MetaUtils v${project.version}
  EcoPower vDEV - 5 (git 5c1ecd68)
  MetaGuide v0
  WorldEditSlimefun vUNOFFICIAL
  Netheopoiesis vMODIFIED
  DailyRewards v1.0-SNAPSHOT
  Aircraft v0

### 🧭 Other plugins

Afaik not related to other plugins

**Comment by JustAHuman-xD** [comment]
Can confirm if an outside source cancels either the PlayerInteractEvent or the PlayerRightClickEvent, it does not care and still fires both items. I've made a patch solution in a separate plugin that just closes the backpack when its dropped, idk how we'd fix this specifically

**Comment by WalshyDev** [comment]
I'm starting to like backpacks less and less...

**Comment by J3fftw1** [comment]
Reproductions steps are pretty unclear. 
If i right click in the inventory nothing happens.

**Comment by JustAHuman-xD** [comment]
> Reproductions steps are pretty unclear. If i right click in the inventory nothing happens.

By right click I mean right click to open the inventory, after that you can see the behavior, you can put the backpack in it self, you can drop the backpack on the floor you can do literally anything you are not supposed to be able to do.

**Comment by J3fftw1** [comment]
This then seems fixed in the latest version i cant repro it

**Comment by JustAHuman-xD** [comment]
> This then seems fixed in the latest version i cant repro it

latest as in the PR you made or wdym?

**Comment by JustAHuman-xD** [comment]
just tested on latest SF build and is still broken

**Comment by WalshyDev** [comment]
Can you test with https://github.com/Slimefun/Slimefun4/pull/4134?

**Comment by JustAHuman-xD** [comment]
sure lemme try

**Comment by JustAHuman-xD** [comment]
Can confirm not fixed on latest build, tried with addons and without and both times the bug worked

**Comment by MAG10d** [comment]
Paper git-Paper-496 (MC: 1.20.4)
Slimefun Dev - 1145
Java 17

i cant repro it

**Comment by JustAHuman-xD** [comment]
I'll try again after school using the latest paper builds and I'll let you know if it's still an issue

**Comment by croudom10** [comment]
Couldn't reproduce it in 1148 1.20.6, I believe it has been patched.

**Comment by JustAHuman-xD** [comment]
> I'll try again after school using the latest paper builds and I'll let you know if it's still an issue

I guess I didn't update but when I did this I believe the issue still occurred, I'll check again shortly though

**Comment by JustAHuman-xD** [comment]
Yeah this is still an issue, atm on the latest paper 1.20.4, opening with either setup (bag in offhand or guide in offhand), the guide opens, then the bag opens, and all normal restrictions are gone. I can put other bags and shulker boxes and stuff in the bag, etc
